src/lib.rs

```diff
@@ -144,19 +147,114 @@ mod tests {
         sscanf!(input, "{}{}", _word1, _word2).unwrap();
     }
 
-    #[test]
-    #[should_panic]
-    fn wrong_format_placeholder_type() {
-        let input = "5 -> 5.0";
-        let mut _request: i32 = 0;
-        let mut _reply: f32 = 0.0;
-        sscanf!(input, "{u64} -> {f64}", _request, _reply).unwrap();
-    }
-
     #[test]
     fn into_array_elements() {
         let s = "3,4";
         let mut arr: [f64; 2] = [0.0; 2];
         sscanf!(&s, "{},{}", arr[0], arr[1]).unwrap();
     }
+
+    #[test]
+    fn test_variable_names_basic() {
+        let input = "John: 25";
+        let mut name: String = String::new();
+        let mut age: i32 = 0;
+
+        // Using variable names instead of types in format string
+        sscanf!(input, "{name}: {age}", name, age).unwrap();
+        assert_eq!(name, "John");
+        assert_eq!(age, 25);
+    }
+
+    #[test]
+    fn test_mixed_variable_names_and_generic() {
+        let input = "Temperature: 23.5 degrees";
+        let mut location: String = String::new();
+        let mut temp: f32 = 0.0;
+        let mut unit: String = String::new();
+
+        // Mix variable names and generic placeholders
+        sscanf!(input, "{location}: {} {unit}", location, temp, unit).unwrap();
+        assert_eq!(location, "Temperature");
+        assert_eq!(temp, 23.5);
+        assert_eq!(unit, "degrees");
+    }
```

Comment:
Another test like this but that variables does not follow same order in params like in the format string could be interesting.